sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <chrono>
+#include "opentelemetry/core/timestamp.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/trace/canonical_code.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * SpanData is a representation of all data collected by a span.
+ */
+class SpanData final : public Recordable
+{
+public:
+  /**
+   * Initialized all span fields with default values. Time stamps and duration
+   * are initialized with 0, the status of the span is initialized with OK.
+   */
+  SpanData() noexcept : duration_(0), status_code_(opentelemetry::trace::CanonicalCode::OK) {}
+
+  /**
+   * Get the trace id for this span
+   * @return the trace id for this span
+   */
+  opentelemetry::trace::TraceId GetTraceId() const noexcept { return trace_id_; }
+
+  /**
+   * Get the span id for this span
+   * @return the span id for this span
+   */
+  opentelemetry::trace::SpanId GetSpanId() const noexcept { return span_id_; }
+
+  /**
+   * Get the parent span id for this span
+   * @return the span id for this span's parent
+   */
+  opentelemetry::trace::SpanId GetParentSpanId() const noexcept { return parent_span_id_; }
+
+  /**
+   * Get the name for this span
+   * @return the name for this span
+   */
+  opentelemetry::nostd::string_view GetName() const noexcept { return name_; }
+
+  /**
+   * Get the status for this span
+   * @return the status for this span
+   */
+  opentelemetry::trace::CanonicalCode GetStatus() const noexcept { return status_code_; }
+
+  /**
+   * Get the status description for this span
+   * @return the description of the the status of this span
+   */
+  opentelemetry::nostd::string_view GetDescription() const noexcept { return status_desc_; }
+
+  /**
+   * Get the start time for this span
+   * @return the start time for this span
+   */
+  opentelemetry::core::SystemTimestamp GetStartTime() const noexcept { return start_time_; }
+
+  /**
+   * Get the duration for this span
+   * @return the duration for this span
+   */
+  std::chrono::nanoseconds GetDuration() const noexcept { return duration_; }
+
+  void SetTraceId(opentelemetry::trace::TraceId trace_id) noexcept { trace_id_ = trace_id; }
+
+  void SetSpanId(opentelemetry::trace::SpanId span_id) noexcept { span_id_ = span_id; }
+
+  void SetParentSpanId(opentelemetry::trace::SpanId parent_span_id) noexcept
+  {
+    parent_span_id_ = parent_span_id;
+  }
+
+  void AddEvent(nostd::string_view name, core::SystemTimestamp timestamp) noexcept
+  {
+    (void)name;
+    (void)timestamp;
+  }
+
+  void SetStatus(trace_api::CanonicalCode code, nostd::string_view description) noexcept
```

Comment:
Can you add override for virtual methods the derived class implements?

Reply:
Done.